### PR TITLE
Streamline dockerfile, use cache

### PR DESCRIPTION
**Pull request by tomaszdurka**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
-RUN npm run build
+FROM node:16-alpine3.17 as builder
+WORKDIR /build
+COPY . .
+RUN rm -rf .env*
+COPY --from=dependencies /dependencies .
+RUN npm run build --prod
 
+
+FROM node:16-alpine3.17
```
Comment (Itrulia) [inline Dockerfile:22]: couldnt we use a slim package here?

Comment (Itrulia) [reply Dockerfile:22]: We also miss cleaning up the apt-get cache, should be a few hundred MBs too, no?

File: Dockerfile [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
-RUN npm run build
+FROM node:16-alpine3.17 as builder
+WORKDIR /build
+COPY . .
+RUN rm -rf .env*
+COPY --from=dependencies /dependencies .
+RUN npm run build --prod
 
+
+FROM node:16-alpine3.17 as runtime
```
Comment (Itrulia) [inline Dockerfile:22]: Using `FROM gcr.io/distroless/nodejs:16` here would save 80MB

File: FAQ.md
Comment (Itrulia) [inline FAQ.md:1]: Hm those changes look not quite right

File: Dockerfile (hunk 1 above)
Comment (tomaszdurka) [reply Dockerfile]: > couldnt we use a slim package here?

Size was 2mb diff and actually npm ci command took longer 250 slim vs 150 alpine secs locally...

File: Dockerfile (hunk 2 above)
Comment (tomaszdurka) [reply Dockerfile:18]: > Using `FROM gcr.io/distroless/nodejs:16` here would save 80MB

```
#12 0.276 container_linux.go:380: starting container process caused: exec: "/bin/sh": stat /bin/sh: no such file or directory
------
executor failed running [/bin/sh -c npm ci]: exit code: 1
```

I get this on distroless so wont't go there atm.

File: FAQ.md
Comment (tomaszdurka) [reply FAQ.md:1]: That was 1 liner change to trigger build ...
Reformatted whole thing too - reverting.

**Comment by elias-summermatter** [comment]
@tomaszdurka  Do we still need this? Merge or close.

**Comment by Itrulia** [comment]
@elias-summermatter  still something we want to do and got stuck because of `bcrypt` as `bcrypt` doesn't allow us to bundle the whole application. I have a few other things I wanna try but didn't get to do it yet.

**Pull request closed without merging**